.github/workflows/reusable-acceptance.yml

```diff
@@ -26,40 +25,50 @@ on:
       kind-version:
         required: false
         type: string
-        default: "v1.22.4"
+        default: "v1.24.6"
+      checkout-ref:
+        required: false
+        type: string
+        default: ${{ github.sha }}
     secrets:
         CONSUL_ENT_LICENSE:
           required: true
 
 # Environment variables can only be used at the step level
 env:
   TEST_RESULTS: /tmp/test-results # path to where test results are saved
-  CONSUL_ENT_LICENSE: ${{ secrets.CONSUL_ENT_LICENSE }} 
+  CONSUL_ENT_LICENSE: ${{ secrets.CONSUL_ENT_LICENSE }}
+  CONSUL_K8S_IMAGE: ${{ inputs.consul-k8s-image }}
 
 jobs:
   job:
-    runs-on: ubuntu-latest
+    runs-on: [custom, linux, xl]
     strategy: 
       matrix:
-        include: # I am really sorry for this but I could not find a way to automatically split our tests into several runners. For now, split manually.
-          - {runner: "0", test-packages: "basic connect consul-dns"}
-          - {runner: "1", test-packages: "controller example ingress-gateway"}
-          - {runner: "2", test-packages: "mesh-gateway metrics"}
-          - {runner: "3", test-packages: "partitions sync terminating-gateway"}
-          - {runner: "4", test-packages: "vault"}
+        include:
+          - {runner: "0", test-packages: "basic"}
+          - {runner: "0", test-packages: "basic consul-dns"}
+          - {runner: "1", test-packages: "connect"}
+          - {runner: "2", test-packages: "controller example"}
+          - {runner: "3", test-packages: "ingress-gateway"}
+          - {runner: "4", test-packages: "metrics partitions"}
+          - {runner: "5", test-packages: "peering"}
+          - {runner: "6", test-packages: "snapshot-agent sync terminating-gateway vault wan-federation"}
 
-      fail-fast: true
+      fail-fast: false
```

Comment:
I think we'd want to fail-fast, right?

Reply:
So if we don't fail fast, it lets all the other acceptance tests run through to completion. Which is great if the rest of your tests pass. Then you can `rerun failed` and only re-run the failed cases (i.e. rerun just the connect tests)

Reply:
Ahhh ok! That's a good point.